### PR TITLE
MAINT, DOC: fix new typos detected by codespell

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**

### CODE REVIEW

File: numpy/core/tests/test_api.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def test_array_array():
 
 @pytest.mark.parametrize("array", [True, False])
 def test_array_impossible_casts(array):
-    # All builtin types can forst cast as least theoretically
+    # All builtin types can forst cast at least theoretically
```
Comment (charris) [inline numpy/core/tests/test_api.py:147]: This still needs work. I think it is supposed to be
```
# All builtin types can be force cast,  at least theoretically,
```
@seberg Thoughts?

File: numpy/lib/npyio.py [hunk 2]
```diff
@@ -818,7 +818,7 @@ def _loadtxt_pack_items(packing, items):
 
 def _ensure_ndmin_ndarray_check_param(ndmin):
     """Just checks if the param ndmin is supported on
-        _ensure_ndmin_ndarray. Is intented to be used as
+        _ensure_ndmin_ndarray. Is intended to be used as
```
Comment (charris) [inline numpy/lib/npyio.py:821]: "It is intended..."

**Comment by charris** [comment]
Couple of nits, otherwise good.

File: numpy/core/tests/test_api.py (hunk 1 above)
Comment (seberg) [reply numpy/core/tests/test_api.py:147]: Yes, the commas are indeed missing, thanks!

Comment (DimitriPapadopoulos) [reply numpy/core/tests/test_api.py:147]: Ah thank you. I had been unable to find an alternative that makes sense - I had tried "_first cast_" but it didn't make sense. Perhaps "_can be forcibly cast_"?

Comment (seberg) [reply numpy/core/tests/test_api.py:147]: oh, hadn't noticed the "forst" :)

Comment (DimitriPapadopoulos) [reply numpy/core/tests/test_api.py]: Fixed, thank you.

File: numpy/lib/npyio.py (hunk 2 above)
Comment (DimitriPapadopoulos) [reply numpy/lib/npyio.py]: Fixed too.

**Comment by charris** [comment]
Thanks @DimitriPapadopoulos .

**Comment by ganesh-k13** [comment]
Hey all, was wondering if #19594 will help in these cases.

**Pull request merged**